app/models/site_configuration.rb

```diff
@@ -10,6 +13,34 @@ def self.instance
 
   # A private method to ensure only one record exists
   def ensure_single_record
-    errors.add(:base, "There can only be one SiteConfiguration record") if SiteConfiguration.count > 0
+    if SiteConfiguration.count > 0
+      errors.add(:base, I18n.t("activerecord.errors.models.site_configuration.single_record"))
+    end
+  end
+
+  def validate_help_link_items
+    return unless help_link_items.present?
+
+    help_link_items.each do |key, item|
+      # Check if item should show
+      if item["show"]
+        unless item["href"].present? && item["title"].present? && item["description"].present?
+          errors.add(
+            :base,
+            I18n.t("activerecord.errors.models.site_configuration.attributes.help_link_items.incomplete", link: key),
+          )
+        end
+      end
+
+      # Check if href is a valid URL
+      if item["href"].present?
+        unless item["href"].match?(%r{\Ahttps://.+\z})
+          errors.add(
+            :base,
+            I18n.t("activerecord.errors.models.site_configuration.attributes.help_link_items.invalid_url", link: key),
+          )
```

Comment:
can you follow the same method as this https://github.com/bcgov/HOUS-permit-portal/blob/develop/app/models/concerns/validate_url_attributes.rb . You can limit to https, but prefer this over regex